### PR TITLE
Added functionality

### PR DESCRIPTION
**Pull request by SimonBarrettACT**
Needed to make a couple of small changes for a project and thought that others might find these useful:

*   A config setting to set the input to be displayed full width
*   Added the ability to hide the add note input (can be displayed depending on user role/permissions)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ This [Laravel Nova](https://nova.laravel.com) package adds a notes field to Nova
 - Ability to add notes through the UI or programmatically
 - Ability to delete user-made notes (w/ confirmation modal)
 - Customizable placeholder support
+- Set abilty to hide or show the add note input
```
Comment (Tarpsvo) [inline README.md:12]: - Set abilty to hide or show the add note button

File: src/NotesField.php [hunk 2]
```diff
@@ -10,14 +10,66 @@ class NotesField extends Field
...
+
+    /**
+     * Sets the input size.
+     *
+     * @param string $inputWidth
+     * @return NotesField
+     **/
+    public function inputWidth($inputWidth)
+    {
+        return $this->withMeta(['inputWidth' => $inputWidth]);
```
Comment (Tarpsvo) [inline src/NotesField.php:50]: Is this used somewhere?

File: src/NotesField.php [hunk 3]
```diff
@@ -10,14 +10,66 @@ class NotesField extends Field
...
+
+    /**
+     * Show or hide the notes section
+     *
+     * @param string $showNotes
+     * @return NotesField
+     **/
+    public function showNotes($showNotes)
+    {
+        return $this->withMeta(['showNotes' => $showNotes]);
```
Comment (Tarpsvo) [inline src/NotesField.php:72]: This doesn't seem to be used in the JS either. What should it do, hide the notes field altogether?

File: src/NotesField.php [hunk 4]
```diff
@@ -10,14 +10,66 @@ class NotesField extends Field
...
+        return $this->withMeta(['inputWidth' => $inputWidth]);
+    }
+
+    /**
+     * Show or hide the AddNote input.
+     *
+     * @param string $showAddNote
+     * @return NotesField
+     **/
+    public function showAddNote($showAddNote)
```
Comment (Tarpsvo) [inline src/NotesField.php:59]: Can this be renamed to showAddNoteButton please?

**Review by Tarpsvo — CHANGES_REQUESTED** [review]

**Comment by Tarpsvo** [comment]
Hi! Thanks a lot for the contribution! I added some comments regarding the code. There seemed to be some unused functions. I also requested some wording changes, if that's okay for you. Thanks again!

**Comment by SimonBarrettACT** [comment]
Thanks,

I'll take a look and make the changes you've requested.

Cheers
Simon




Simon Barrett
Digital Implementation Officer


W:      www.acttraining.org.uk<http://www.acttraining.org.uk>

[https://i.imgur.com/rwoIqP5.jpg]
[http://i.imgur.com/HelM7nf.png]<https://www.facebook.com/acttraining/>    [http://i.imgur.com/rgbz5t5.png] <https://twitter.com/acttrainingltd>     [http://i.imgur.com/aI1oh53.png] <https://www.instagram.com/acttraining/>     [http://i.imgur.com/MiQ4PGb.png] <https://www.youtube.com/channel/UCVuxUkvWdcKxEvIMt4IphKQ>

[http://i.imgur.com/NLp6GkU.jpg]<https://www.ntfw.org/events/apprenticeships-in-wales-meet-the-training-providers/>

ACT is an environmentally-aware business ... please don't print this e-mail unless you really need to.

This E-Mail message together with any attachments is intended solely for the recipients as named; the E-Mail may contain material which is confidential and privileged.


________________________________

File: README.md (hunk 1 above)
Comment (SimonBarrettACT) [reply README.md:12]: I've changed it to:

Set ability to hide or show the 'Add Note' button

File: src/NotesField.php (hunk 2 above)
Comment (SimonBarrettACT) [reply src/NotesField.php:50]: I forgot to remove this when changing from a function to a config setting.

File: src/NotesField.php (hunk 4 above)
Comment (SimonBarrettACT) [reply src/NotesField.php:59]: No problem, I'll also update the readme to reflect this.

File: src/NotesField.php (hunk 3 above)
Comment (SimonBarrettACT) [reply src/NotesField.php:72]: Sorry, I was playing with the idea of having an option to hide notes for roles/permissions that do not have permission to view them. Meant to remove this before pushing.

I can put it back in if you think it's useful.

**Comment by SimonBarrettACT** [comment]
I'm also thinking that it would be useful to provide the ability to edit notes. What do you think?

Have now made and pushed the requested changes.

It's a great custom field, really useful.

**Comment by Tarpsvo** [comment]
Hi! I merged it and released it in version 1.5.5.

However, I did make further changes.

1) The config option `use_full_width` became `full_width_inputs`
2) `->fullWidth(true/false)` can now be called on a field instance, allowing you to make only some fields full width
3) `showAddNoteButton()` became `addingNotesEnabled()`

That should be it. Thanks a ton for your help and good luck!

**Pull request merged**